sdk/src/suins-transaction.ts

```diff
@@ -0,0 +1,315 @@
+			throw new Error('Subdomains package ID not found');
+		if (isParentSubdomain && !this.#suinsClient.constants.tempSubNamesProxyPackageId)
+			throw new Error('Subdomains proxy package ID not found');
+
+		this.transactionBlock.moveCall({
+			target: isParentSubdomain
+				? `${this.#suinsClient.constants.tempSubNamesProxyPackageId}::subdomain_proxy::new_leaf`
+				: `${this.#suinsClient.constants.subNamesPackageId}::subdomains::new_leaf`,
+			arguments: [
+				this.transactionBlock.object(this.#suinsClient.constants.suinsObjectId),
+				this.transactionBlock.object(parentNft),
+				this.transactionBlock.object(SUI_CLOCK_OBJECT_ID),
+				this.transactionBlock.pure.string(normalizeSuiNSName(name, 'dot')),
+				this.transactionBlock.pure.address(targetAddress),
+			],
+		});
+	}
+
+	/**
+	 * Removes a leaf subname.
+	 */
+	removeLeafSubName({ parentNft, name }: { parentNft: ObjectArgument; name: string }) {
+		if (!isValidSuiNSName(name)) throw new Error('Invalid SuiNS name');
+		const isParentSubdomain = isNestedSubName(name);
+		if (!isSubName(name)) throw new Error('This can only be invoked for subnames');
+		if (!this.#suinsClient.constants.suinsObjectId) throw new Error('SuiNS Object ID not found');
+		if (!this.#suinsClient.constants.subNamesPackageId)
+			throw new Error('Subdomains package ID not found');
+		if (isParentSubdomain && !this.#suinsClient.constants.tempSubNamesProxyPackageId)
+			throw new Error('Subdomains proxy package ID not found');
+
+		this.transactionBlock.moveCall({
+			target: isParentSubdomain
+				? `${this.#suinsClient.constants.tempSubNamesProxyPackageId}::subdomain_proxy::remove_leaf`
+				: `${this.#suinsClient.constants.subNamesPackageId}::subdomains::remove_leaf`,
+			arguments: [
+				this.transactionBlock.object(this.#suinsClient.constants.suinsObjectId),
+				this.transactionBlock.object(parentNft),
+				this.transactionBlock.object(SUI_CLOCK_OBJECT_ID),
+				this.transactionBlock.pure.string(normalizeSuiNSName(name, 'dot')),
+			],
+		});
+	}
+
+	setTargetAddress({
+		nft,
+		address,
+		isSubdomain,
+	}: {
+		nft: ObjectArgument;
+		address?: string;
+		isSubdomain?: boolean;
+	}) {
+		if (!this.#suinsClient.constants.suinsObjectId) throw new Error('SuiNS Object ID not found');
+		if (!this.#suinsClient.constants.utilsPackageId) throw new Error('Utils package ID not found');
+
+		if (isSubdomain && !this.#suinsClient.constants.tempSubNamesProxyPackageId)
+			throw new Error('Subdomains proxy package ID not found');
+
+		this.transactionBlock.moveCall({
+			target: isSubdomain
+				? `${this.#suinsClient.constants.tempSubNamesProxyPackageId}::subdomain_proxy::set_target_address`
+				: `${this.#suinsClient.constants.utilsPackageId}::direct_setup::set_target_address`,
+			arguments: [
+				this.transactionBlock.object(this.#suinsClient.constants.suinsObjectId),
+				this.transactionBlock.object(nft),
+				this.transactionBlock.pure(address ? [address] : []),
```

Comment:
this should be updated to not use pure this way.  Probably `this.transactionBlock.pure(bcs.vector(bcs.string()).serialize(address ? [address] : []))`

Reply:
@hayes-mysten this is an `Option<address>` in this case; Wen `txb.pure.option()` :eyes: haha

Reply:
`this.transactionBlock.pure(bcs.vector(bcs.string()).serialize(address ? [address] : []))` will still work, but:

but you can do `this.transactionBlock.pure(bcs.option(bcs.string()).serialize(address ?? null))`

Reply:
Options are enums, and enums are encoded with each enum type getting an index, followed by the bytes for the value.

So None is encoded as `[0]`, and and Some is encoded as `[1,...bytesForValue]`, vectors are encoded as `[length, ...bytesForValues]`, so a Vector of length `0` is encoded as `[0]` and a vector of length 1 is encoded as `[1, ...bytesForFirstValue]`